src/daemons/rhsmcertd.c

```diff
@@ -223,15 +223,19 @@ signal_handler(int signo) {
 int
 get_lock ()
 {
-    int fdlock;
+    int fd_lock = open (LOCKFILE, O_WRONLY | O_CREAT, 0640);
+    int ret = 0;
 
-    if ((fdlock = open (LOCKFILE, O_WRONLY | O_CREAT, 0640)) == -1)
-        return 1;
-
-    if (flock (fdlock, LOCK_EX | LOCK_NB) == -1)
-        return 1;
+    if (fd_lock == -1) {
+        ret = 1;
+    } else {
+        if (flock (fd_lock, LOCK_EX | LOCK_NB) == -1) {
```

Comment:
Won't this immediately close the lock even though rhsmcertd is continuing to run? I believe the lock needs to be held until rhsmcertd is killed.

Reply:
I'm sorry. You are right. The lock is released, when corresponding file descriptor has been closed. I will update the PR.